test_tortoise_interpreter.py

```diff
@@ -58,22 +67,28 @@ def test_sanitized_programs(program, expected):
 
 
 def test_navigate_command_functions():
-    def pen_up(command):
-        return f'PEN UP'
-    
-    def pen_colour(command):
-        return f'P {command[1]}'
-
-    COMMANDS = {'U': pen_up, 'P': pen_colour}
-    program = [['P', '3'], ['U']]
-    command_to_parse = navigate_command_functions(COMMANDS, program)
 
-    assert command_to_parse == ['1. P 3', '2. PEN UP']
+    COMMANDS = {'U': pen_up,'D': pen_down ,'P': pen_colour, 'N': move_north, 'E': move_east}
```

Comment:
We shouldn't use the real handler functions here. This ties our test to an implementation detail, and is unnecessary. Implementing test functions shows that the function really is using the commands that are passed in; there's no hard-coding going on.